SOC-812 Email update: article comment 'view all comment' link is broken

rafalkalinski opened the pull request:
https://wikia-inc.atlassian.net/browse/SOC-812

dianafa committed:
init

Commit:
ADEN-1978 INCONTENT_* ads refactoring

Commit:
trending-articles | reorganized code, cleaned up response, made it work

hakubo committed:
Missing indentation

hakubo committed:
type in length

Commit:
trending-articles | renamed the excepion message

Commit:
ADEN-1978 CR fixes

Commit:
trending-articles | use global

Commit:
SOC-815 Added cache busting and cleaned up the function slightly.

Commit:
CONCF-505 | added methods to Mercury API for fetching trending videos

Commit:
SOC-769 remove DISTINCT from query

Commit:
add caption to the parser

Commit:
extend tests

Commit:
ADEN-1978 Use wgArticleId % 2 as a 300x600 or 300x250 decider

Commit:
add caption to image template

Commit:
change logic for getting name and key values to match with default mobile image rendering logic

Commit:
DAT-2812 add custom theming

Commit:
styling image captions inside infobox

fraszczakszymon committed:
Krux segments count tracking

Commit:
SOC-769 remove trailing comma

bkoval committed:
Clearing encrypted access_token cookie on logout

bkoval committed:
Adding parentheses

Commit:
SOC-742 Don't track sign-up events for mirrored requests

Commit:
SOC-742 Move logic to Track class

bkoval committed:
Whitespacing

Commit:
SOC-742 Address code review feedback

Commit:
SOC-736 add support for FB Page Plugin

Commit:
remove html escaping of data values

Commit:
DAT-2811 make theme class appear only when set, theme class will be prefixed instead of suffixed

Commit:
add htnl escaping before rendering + add additional tests for render service

Commit:
ADEN-1978 Detect sections that are too short for ads

Commit:
ADEN-1978 Remove condition we don't need anymore

Commit:
SOC-797 | Mercury logout - code polishing after CR

Commit:
SOC-736 remove extra parens, rename file, update variable/method names

Commit:
trending-articles | simplified the item response

Commit:
CONCF-505 | unify getting trending data

macbre committed:
removeTempUser - initial commit

macbre committed:
RemoveTempUserAccounts: run stylize.php

macbre committed:
RemoveTempUserAccounts: DB profiling

macbre committed:
RemoveTempUserAccounts: check secondary clusters

Commit:
initial commit; adding access to parent; checking parent type

Commit:
SOC-815 Remove conditional in return

Commit:
SOC-815 First part of merge from dev

Commit:
SOC-500  | Fix missing template for captcha

Commit:
ADEN-1773 Test Loading Incremental Ads on long pages

Commit:
Refactor and add top level items to the result

Commit:
Clean up a bit

Commit:
SOC-812 title property renamed to pageTitle in CommentController class

Commit:
Merge remote-tracking branch 'origin/dev' into SOC-812

macbre committed:
RemoveTempUserAccounts: sleep after processing each batch

macbre committed:
LoginForm: get rid of TempUser handling

Commit:
DAT-2811 make sure there are no whitespaces in class name

Commit:
ADEN-1978 Work around IE not having repeat function on strings

Commit:
tests

Commit:
ADEN-2049 Make Exitstitial.js only requested when Ad Engine ext is on

kvas-damian commented:
LGTM

Commit:
ADEN-2049 Clean up: we no longer need MercuryApiModelTest::testAdsContext

Commit:
ADEN-2049 Comment update

Commit:
added cluster g to script

Grunny committed:
(INT-82) Add isMainPage data to getArticle response

Add isMainPage data to getArticle response to reliably check if the
current article is the main page in Mercury.

garthwebb commented on extensions/wikia/Email/Controller/CommentController.class.php:
```diff
@@ -9,7 +9,7 @@
 
 	/** @var \Title */
-	protected $title;
+	protected $pageTtitle;
```
This variable is spelled incorrectly (two t's).

garthwebb commented on extensions/wikia/Email/Controller/CommentController.class.php:
```diff
@@ -78,17 +78,17 @@ public function body() {
 			'editorUserName' => $this->getCurrentUserName(),
 			'editorAvatarURL' => $this->getCurrentAvatarURL(),
 			'summary' => $this->getSummary(),
-			'details' => $this->getDetails(),
```
Ok, it took me a long time to figure out *this* was the change that addressed the bug for SOC-812.  It seems that calling ```getDetails``` has a side effect that screws up creating the link in ```getCommentSectionLink```.  My recommendation would be:

1) Document in a comment that the 'details' line is position dependent so others know not to touch it
2) Realize this is a terrible state of things
3) Erase the comment, move the line back up and find the root cause of the problem ;)

I found the issue, details below...

kvas-damian replied:
Actually this line change has nothing to do right now.
It was a problem when we had variable called ```title``` because there is some strange code that is modifying title parameter in a controller. After we renamed it everything should work ok

garthwebb commented on extensions/wikia/Email/Controller/CommentController.class.php:
```diff
@@ -104,7 +104,7 @@ protected function getSalutation() {
+		$articleTitle = $this->pageTtitle->getText();
 
 		return wfMessage( $this->getSummaryKey(), $articleTitle )
 			->inLanguage( $this->targetLang )
```
Ok, I can't comment on this line because its not in the diff, but a few lines below on L117 we call ```newFromTitle```.  This ends up calling (via hooks) ```ArticleCommentList::ArticleFromTitle```.  This sets the fragment ```#comm-XXX``` and _somehow_ this persists and is applied to the next Title object created.  I'm not positive but I think its because it issues a redirect for some strange reason.

The solution is to not use ```newFromTitle``` which requires this request context.  I think this context is what screws things up.  We can also easily get the article via ID like this:
```php
$article = \Article::newFromID( $this->commentTitle->getArticleID() );
```

rafalkalinski replied:
Like Damian said above - the reason of the bug was the $title property name, and now it is fixed.
Since this line was not causing the problem, I'm gonna leave it like it is.

Commit:
Fixes after merge

macbre committed:
RemoveTempUserAccounts: run per cluster

Commit:
PLATFORM-1234 make PBLoginForm::mainLoginForm() compatible with LoginForm::mainLoginForm()

Commit:
ADEN-2049: removed unnecessary white space

Commit:
simplify

siebrand committed:
Localisation updates from https://translatewiki.net.

Commit:
comments

siebrand committed:
Localisation updates from https://translatewiki.net.

This is the final localisation update from translatewiki.net. It
is a full export/rebuild of supported extensions. So long, and
thanks for all the fish.

Commit:
fix portable infobox render service tests

Grunny committed:
(MAIN-4620) Sanitize custom theme class in PortableInfobox

Sanitize the custom theme class users can provide in Portable Infoboxes.
Sanitizer::escapeClass only allows valid characters in a class name
through.

Commit:
SOC-812 misspell in variable name fixed

Commit:
Merge remote-tracking branch 'origin/dev' into SOC-812

Commit:
Merge remote-tracking branch 'origin/dev' into SOC-812
